tests/ops/qutrit/test_qutrit_matrix_ops.py

```diff
@@ -283,9 +283,363 @@ def test_matrix_representation(self):
+
+    def test_no_control(self):
+        """Test if ControlledQutritUnitary raises an error if control wires are not specified"""
+        with pytest.raises(ValueError, match="Must specify control wires"):
+            qml.ControlledQutritUnitary(U_thadamard_01, wires=2)
+
+    def test_shared_control(self):
+        """Test if ControlledQutritUnitary raises an error if control wires are shared with wires"""
+        with pytest.raises(ValueError, match="The control wires must be different from the wires"):
+            qml.ControlledQutritUnitary(U_thadamard_01, control_wires=[0, 2], wires=2)
+
+    def test_wrong_shape(self):
+        """Test if ControlledQutritUnitary raises a ValueError if a unitary of shape inconsistent
+        with wires is provided"""
+        with pytest.raises(ValueError, match=r"Input unitary must be of shape \(3, 3\)"):
+            qml.ControlledQutritUnitary(np.eye(9), control_wires=[0, 1], wires=2).matrix()
+
+    def test_arbitrary_multiqutrit(self):
+        """Test if ControlledQutritUnitary applies correctly for a 2-qutrit unitary with 2-qutrit
+        control, where the control and target wires are not ordered."""
+        control_wires = [1, 3]
+        target_wires = [2, 0]
+
+        # pick some random unitaries (with a fixed seed) to make the circuit less trivial
+        U1 = unitary_group.rvs(81, random_state=1)
+        U2 = unitary_group.rvs(81, random_state=2)
+
+        # the two-qutrit unitary
+        U = unitary_group.rvs(9, random_state=3)
+
+        # the 4-qutrit representation of the unitary if the control wires were [0, 1] and the target
+        # wires were [2, 3]
+        U_matrix = np.eye(81, dtype=np.complex128)
+        U_matrix[72:81, 72:81] = U
+
+        # We now need to swap wires so that the control wires are [1, 3] and the target wires are
+        # [2, 0]
+        swap = TSWAP
+
+        # initial wire permutation: 0123
+        # target wire permutation: 1320
+        swap1 = np.kron(swap, np.eye(9))  # -> 1023
+        swap2 = np.kron(np.eye(9), swap)  # -> 1032
+        swap3 = np.kron(np.kron(np.eye(3), swap), np.eye(3))  # -> 1302
+        swap4 = np.kron(np.eye(9), swap)  # -> 1320
+
+        all_swap = swap4 @ swap3 @ swap2 @ swap1
+        U_matrix = all_swap.T @ U_matrix @ all_swap
+
+        dev = qml.device("default.qutrit", wires=4)
+
+        @qml.qnode(dev)
+        def f1():
+            qml.QutritUnitary(U1, wires=range(4))
+            qml.ControlledQutritUnitary(U, control_wires=control_wires, wires=target_wires)
+            qml.QutritUnitary(U2, wires=range(4))
+            return qml.state()
+
+        @qml.qnode(dev)
+        def f2():
+            qml.QutritUnitary(U1, wires=range(4))
+            qml.QutritUnitary(U_matrix, wires=range(4))
+            qml.QutritUnitary(U2, wires=range(4))
+            return qml.state()
+
+        state_1 = f1()
+        state_2 = f2()
+
+        assert np.allclose(state_1, state_2)
```

Comment:
I feel like this test needs a test :sweat_smile: I was able to follow though, and this is a good sanity check to make sure things work under wire reordering.